src/Metrics/DefaultMetricsSender.php

```diff
@@ -31,12 +31,15 @@ public function sendMetrics(MetricsBucket $bucket): void
                 'instanceId' => $this->configuration->getInstanceId(),
                 'bucket' => $bucket->jsonSerialize(),
             ], JSON_THROW_ON_ERROR)));
+        if ($this->configuration->getProxyKey() !== null) {
+            $request = $request->withHeader('Authorization', $this->configuration->getProxyKey());
+        }
```

Comment:
Add the same todo comment here, please.

Reply:
I'm going to have one more stab at the builder before converting this to a proper PR. Thanks for the feedback :+1: